renku/api/datasets.py

```diff
@@ -110,8 +119,75 @@ def with_dataset(self, name=None):
             #     if path.exists():
             #         raise ValueError('Dataset already exists')
 
-            with path.open('w') as f:
-                yaml.dump(source, f, default_flow_style=False)
+            if path and path.parent.exists():
+                with path.open('w') as f:
+                    yaml.dump(source, f, default_flow_style=False)
+
+    def delete_dataset(self, dataset, force=False):
+        """Delete empty dataset.
+
+        :raises`errors.DatasetNotEmpty`: If dataset contains files.
+        :param dataset: Dataset instance which we are deleting.
+        :param force: If set to ``True`` it will delete all data
+            from the dataset.
+        """
+        if not force and dataset.files:
+            raise errors.DatasetNotEmpty()
+
+        dataset_dir_path = self.renku_datasets_path / dataset.identifier.hex
```

Comment:
you might need to iterate over all `refs/datasets` since there might be more aliases.

Reply:
we might need a another `renku doctor` check that find broken `refs`